planner/optimize.go

```diff
@@ -162,6 +162,12 @@ func Optimize(ctx context.Context, sctx sessionctx.Context, node ast.Node, is in
 		}
 	}
 
+	// Override the resource group if necessary
+	// TODO: we didn't check the existence of the hinted resource group now to save the cost per query
+	if originStmtHints.HasResourceGroup {
```

Comment:
Do we need to check if the resource control is enabled?

Reply:
check added